staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/streaming.go

```diff
@@ -0,0 +1,249 @@
+	}
+	if listMetaType.Tag.Get("json") != "metadata,omitempty" {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected ListMeta field tag")
+	}
+	listMetaValue := v.FieldByName("ListMeta")
+	if !listMetaValue.IsValid() {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected ListMeta field")
+	}
+	listMeta, ok := listMetaValue.Interface().(metav1.ListMeta)
+	if !ok {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected ListMeta field to have ListMeta type")
+	}
+	typeMetaField, ok := listType.FieldByName("TypeMeta")
+	if !ok {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected TypeMeta field")
+	}
+	if typeMetaField.Tag.Get("json") != ",inline" {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected TypeMeta field tag")
+	}
+	typeMetaValue := v.FieldByName("TypeMeta")
+	if !typeMetaValue.IsValid() {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected TypeMeta field")
+	}
+	typeMeta, ok := typeMetaValue.Interface().(metav1.TypeMeta)
+	if !ok {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected TypeMeta field to have TypeMeta type")
+	}
+
+	itemsField, ok := listType.FieldByName("Items")
+	if !ok {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected Items field")
+	}
+	if itemsField.Tag.Get("json") != "items" {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected TypeMeta field tag")
+	}
+	items, err := meta.ExtractList(list)
+	if err != nil {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, err
+	}
+	return typeMeta, listMeta, items, nil
+}
+
+func streamingEncodeList(w io.Writer, typeMeta metav1.TypeMeta, listMeta metav1.ListMeta, items []runtime.Object) error {
+	_, err := w.Write([]byte(`{`))
+	if err != nil {
+		return err
+	}
+	if typeMeta.Kind != "" {
+		err := encodeValue(w, []byte(`"kind":`), typeMeta.Kind, []byte(","))
+		if err != nil {
+			return err
+		}
+	}
+	if typeMeta.APIVersion != "" {
+		err := encodeValue(w, []byte(`"apiVersion":`), typeMeta.APIVersion, []byte(","))
+		if err != nil {
+			return err
+		}
+	}
+	err = encodeValue(w, []byte(`"metadata":`), listMeta, []byte(`,"items":`))
+	if err != nil {
+		return err
+	}
+	if items == nil {
+		_, err = w.Write([]byte("null}\n"))
+		return err
+	}
+	if len(items) == 0 {
+		_, err = w.Write([]byte("[]}\n"))
+		return err
+	}
+	suffix := []byte(",")
+	prefix := []byte("[")
+	for i, item := range items {
+		if i == len(items)-1 {
+			suffix = nil
+		}
+		err = encodeValue(w, prefix, item, suffix)
+		if err != nil {
+			return err
+		}
+		if i == 0 {
+			prefix = nil
+		}
+	}
+	_, err = w.Write([]byte("]}\n"))
+	return err
+}
+
+func encodeValue(w io.Writer, prefix []byte, value any, suffix []byte) error {
+	data, err := json.Marshal(value)
+	if err != nil {
+		return err
+	}
+	if len(prefix) != 0 {
+		_, err = w.Write(prefix)
+		if err != nil {
+			return err
+		}
+	}
+	_, err = w.Write(data)
```

Comment:
Great idea !

How does the client know how to consume the stream?
What if a client requests a different representation of the list, say a table?